app/assets/javascripts/shared/dradis_datatable.js

```diff
@@ -2,35 +2,129 @@ class DradisDatatable {
-      "<'dataTables_footer_content'ip>",
+      buttons: {
+        dom: {
+          button: {
+            tag: 'button',
+            className: 'btn'
+          }
+        },
+        buttons: [
+          {
+            available: function(){
+              return that.$table.find('td.select-checkbox').length;
+            },
+            attr: {
+              id: 'select-all'
+            },
+            name: 'selectAll',
+            text: '<label for="select-all-checkbox" class="sr-only">Select all"</label><input type="checkbox" id="select-all-checkbox" />',
+            titleAttr: 'Select all'
+          },
+          {
+            extend: 'colvis',
+            text: '<i class="fa fa-columns mr-1"></i><i class="fa fa-caret-down"></i>',
+            titleAttr: 'Choose columns to show',
+            className: 'btn',
+            columns: colvisColumnIndexes
+          }
+        ]
+      },
+      dom: "<'row'<'col-lg-6'B><'col-lg-6'f>>" +
+        "<'row'<'col-lg-12'tr>>" +
+        "<'dataTables_footer_content'ip>",
       initComplete: function (settings) {
         settings.oInstance.wrap("<div class='table-wrapper'></div>");
       },
-      pageLength: 25
+      lengthChange: false,
+      pageLength: 25,
+      select: {
+        selector: 'td:first-child',
+        style: 'multi'
+      }
     });
 
-    // Assign the instantiated DataTable as a DradisDatatable property
-    this.dataTable = dataTable;
     this.behaviors();
   }
 
   behaviors() {
+    this.hideColumns();
+
+    this.setupCheckboxListeners();
+
     this.unbindDataTable();
   }
 
+  hideColumns() {
+    // Hide <th> columns that has data-visible="false"
+    var that = this;
+    that.tableHeaders.forEach(function(column, index) {
+      if (column.dataset.visible == 'false') {
+        var dataTableColumn = that.dataTable.column(index);
+        dataTableColumn.visible(false);
+      }
+    });
+  }
+
   unbindDataTable() {
     var that = this;
 
     document.addEventListener('turbolinks:before-cache', function() {
       that.dataTable.destroy();
     });
   }
+
+
+  ///////////////////// Checkbox /////////////////////
+
+  setupCheckboxListeners() {
+    var that = this,
+        $selectAllBtn = $(this.dataTable.buttons('#select-all').nodes()[0]);
+
+    this.dataTable.on('select.dt deselect.dt', function() {
+      $selectAllBtn.find('#select-all-checkbox').prop('checked', that.areAllSelected());
+
+      if (that.areAllSelected()){
+        $selectAllBtn.attr('title', 'Deselect all');
+      }
+      else {
+        $selectAllBtn.attr('title', 'Select all');
+      }
+    });
+
+    // Remove default datatable button listener to make the checkbox "checking"
+    // work, before adding our own click handler.
+    $selectAllBtn.off('click.dtb').click( function (){
+      if (that.areAllSelected()) {
+        that.dataTable.rows().deselect();
+      }
+      else {
+        that.dataTable.rows().select();
+      }
+    });
```

Comment:
I'm still kinda confused with this, why is this needed?

Reply:
@sean-yeoh we're not using the Datatables' built-in selectAll button since it doesn't fit our use case. That means we have to implement the selectAll/deselectAll functionality ourselves.